Use local peer ledger height from ledger instead of discovery in Gateway

bestbeforetoday opened the pull request:
Discovery ledger heights can be slightly behind the actual peer ledger height. Ensure the most up-to-date information is used for the local peer by querying the ledger height directly. This avoids the possibility of a remote peer with a lower ledger height being selected as an endorser.

Refactored Gateway service method implementations and associated tests into separate files as the size of the api files were getting unmanageable.

Closes hyperledger/fabric-gateway#558

yacovm reviewed (dismissed):
Can we please separate the refactoring part and using the ledger into different pull requests? 

I'm not sure merging one implies the other should be merged and vice versa.

bestbeforetoday commented:
> Can we please separate the refactoring part and using the ledger into different pull requests?

See #4032. I will rebase this change on top of that change.

bestbeforetoday committed:
Use local peer ledger height from ledger instead of discovery in Gateway

Discovery ledger heights can be slightly behind the actual peer ledger height. Ensure the most up-to-date information is used for the local peer by querying the ledger height directly. This avoids the possibility of a remote peer with a lower ledger height being selected as an endorser.

Signed-off-by: Mark S. Lewis <mark_lewis@uk.ibm.com>

yacovm commented on internal/pkg/gateway/registry.go:
```diff
@@ -200,13 +202,42 @@ func (reg *registry) endorsersByOrg(channel string, chaincode string) map[string
+func (reg *registry) localLedgerHeight(channel string) (height uint64, ok bool) {
+	ledger, err := reg.localProvider.Ledger(channel)
+	if err != nil {
+		reg.logger.Debugw("local endorser is not a member of channel", "channel", channel, "err", err)
```
why is this a debug? You are receiving an abnormal request. It should be a warning no?

bestbeforetoday replied:
Sure, I can change it to warning or error. It should never happen

yacovm commented on internal/pkg/gateway/registry.go:
```diff
@@ -200,13 +202,42 @@ func (reg *registry) endorsersByOrg(channel string, chaincode string) map[string
+
+	info, err := ledger.GetBlockchainInfo()
+	if err != nil {
+		logger.Debugw("failed to get local ledger info", "err", err)
```
This should be an error, no?

bestbeforetoday replied:
Pushed an update with the changes to log level

yacovm reviewed (commented):
LGTM modulo comments. Maybe Andrew can take a look at it too.
